### PR TITLE
[SPARK-30350][SQL] Fix ScalaReflection to use an empty array for getting its class object

### PR DESCRIPTION
**Pull request by sekikn**
### What changes were proposed in this pull request?

This PR fixes `ScalaReflection.arrayClassFor()` to use an empty array instead of a one-element array for getting its class object by reflection.

### Why are the changes needed?

Because it may reduce unnecessary memory allocation.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Ran the existing unit tests for sql/catalyst and confirmed that all of them succeeded.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
I guess the code change is OK if the Jenkins build passes.

**Comment by maropu** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala]: cc: @srowen

**Review by HeartSaVioR — APPROVED** [review]
LGTM as tests passed.

I've played with simple code which would verify the change doesn't affect the result.

```
val clazz = classOf[Int]
java.lang.reflect.Array.newInstance(clazz, 1).getClass eq java.lang.reflect.Array.newInstance(clazz, 0).getClass
```

**Review by maropu — APPROVED** [review]
@HeartSaVioR Thanks for the check. LGTM. Leave this to one more committer/PMC.

**Review by srowen — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Pull request closed without merging**
